com.unity.render-pipelines.universal/ShaderLibrary/ShaderVariablesFunctions.hlsl

```diff
@@ -149,17 +149,51 @@ void GetLeftHandedViewSpaceMatrices(out float4x4 viewMatrix, out float4x4 projMa
     projMatrix._13_23_33_43 = -projMatrix._13_23_33_43;
 }
 
-void AlphaDiscard(real alpha, real cutoff, real offset = real(0.0))
+// Only define the alpha clipping helpers when the alpha test define is present.
+// This should help identify usage errors early.
+#if defined(_ALPHATEST_ON)
```

Comment:
Restoring the AlphaDiscard() reduces the impact of this issue a bit, but overall I'm not sure why we would want to expose these functions to users in cases where they cannot function as intended? The previous implementation just turns into a no-op when the alpha test macro isn't defined. This seems like a case of silent failure versus loud error and I usually prefer loud errors myself. :)

If we were to expose these functions, I think their implementation would just always return false for IsAlphaToMaskAvailable(), and always return input alpha for AlphaClip(). Not sure if that would be useful?

Reply:
I agree with you as I mentioned. I like to fail fast too. 

But in the past I've been told to not do this, so that we don't couple our users to our keywords (perhaps due to variant explosion). 
Users can set the constants from code, so it is functional code, but of course they can set keywords too. Or if they fork the shader or SRP they can just add a define.
Let me get another opinion on it and maybe we can agree to change it.

We could also put the Alpha stuff  in it's own file and you just include that if you need Alpha ops.

Reply:
We discussed this internally. It should be ok to land like the way you have it now.